3party/opening_hours/rules_evaluation.cpp

```diff
@@ -385,6 +385,56 @@ bool IsActive(RuleSequence const & rule, time_t const timestamp)
   return res.first && res.second;
 }
 
+time_t GetNextTimeOpen(TRuleSequences const & rules, time_t const dateTime)
+{
+  const time_t TIME_T_MAX = std::numeric_limits<time_t>::max();
+  time_t dateTimeResult = TIME_T_MAX;
+  time_t dateTimeToCheck;
+
+  for (int i = 0; i < 7; i++) // Check in the next 7 days
+  {
+    for (auto it = rules.rbegin(); it != rules.rend(); ++it)
+    {
+      auto const & times = it->GetTimes();
+
+      // If the rule has no times specified, check at 00:00
+      if (times.size() == 0)
+      {
+        // Reset time to 00:00 local time
+        dateTimeToCheck = dateTime - dateTime % (24 * 60 * 60) + i * (24 * 60 * 60) + timezone;
```

Comment:
1. Where does timezone come from?
2. A proper implementation should take timezone as a parameter, and calculate open/close based on the timezone of the place, not based on the timezone of the user. At least we can pass it as a parameter now and write another todo.

Reply:
The timezone was not working properly. I changed it to use MakeTimetuple and mktime instead of doing it by hand. It is always working with local timezone as does the rest of the opening_hours tool.